Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift

```diff
@@ -1014,6 +1014,48 @@ class RenderNodeTranslatorTests: XCTestCase {
         XCTAssertEqual(l.syntax, "swift")
         XCTAssertEqual(l.code, ["func foo() {}"])
     }
+
+    func testNestedSnippetSliceToCodeListing() throws {
+        let (bundle, context) = try testBundleAndContext(named: "Snippets")
+        let reference = ResolvedTopicReference(bundleIdentifier: bundle.identifier, path: "/documentation/Snippets/Snippets", sourceLanguage: .swift)
+        let article = try XCTUnwrap(context.entity(with: reference).semantic as? Article)
+        var translator = RenderNodeTranslator(context: context, bundle: bundle, identifier: reference, source: nil)
+        let renderNode = try XCTUnwrap(translator.visitArticle(article) as? RenderNode)
+        let discussion = try XCTUnwrap(renderNode.primaryContentSections.first(where: { $0.kind == .content }) as? ContentRenderSection)
+        
+        let lastCodeListingIndex = try XCTUnwrap(discussion.content.indices.last {
+            guard case .codeListing = discussion.content[$0] else {
+                return false
+            }
+            return true
+        })
+
+        if case let .paragraph(p) = discussion.content.dropFirst(2).first {
+            XCTAssertEqual(p.inlineContent, [.text("Does a foo.")])
+        } else {
+            XCTFail("Unexpected content where snippet explanation should be.")
+        }
+
+        if case let .codeListing(l) = discussion.content.dropFirst(3).first {
```

Comment:
Are you sure this is actually testing the snippet inside the tab navigator? I _think_ it might just be duplicating the `testSnippetToCodeListing` test above  asserting on the code snippet that's directly above the tab navigator.

You could try changing the markup in `Snippets.md` to reference a different snippet slice to confirm:

```md
    @Tab("hey") {
        Hey there.

        @Small {
            Hey but small.
        }

        @Snippet(path: "Snippets/Snippets/MySnippet", slice: "middle") {}
    }
}
```

That should create a code block that just contains:

```swift
middle()
```